lib/core/src/Cardano/Wallet/Transaction.hs

```diff
@@ -66,3 +91,58 @@ data ErrMkStdTx
     | ErrInvalidTx
     -- ^ when transaction with 0 amount is tried (not valid in Byron)
     deriving (Eq, Show)
+
+-- | Backend-specific variables used by 'estimateMaxNumberOfInputsBase'.
+data EstimateMaxNumberOfInputsParams = EstimateMaxNumberOfInputsParams
+    { estMeasureTx :: [TxIn] -> [TxOut] -> [TxWitness] -> Int
+        -- ^ Finds the size of a serialized transaction.
+    , estAddressSample :: Address -- ^ Address to use in tx output
+    , estBlockHashSize :: Int -- ^ Block ID size
+    , estTxWitnessSize :: Int -- ^ Tx Witness size
+    }
+
+-- | This is called by the 'TransactionLayer' implementation. It uses the
+-- serialization functions to calculate the size of an empty transaction
+-- compared to a transaction with one input. The estimation is based on that.
+--
+-- It doesn't account for transaction outputs, and assumes there is a single Tx
+-- output.
+--
+-- All the values used are the smaller ones. For example, the shortest adress
+-- type and shortest witness type are chosen to use for the estimate.
+estimateMaxNumberOfInputsBase
+    :: EstimateMaxNumberOfInputsParams
+    -- ^ Backend-specific variables used in the estimation
+    -> Quantity "byte" Word16
+    -- ^ Transaction max size in bytes
+    -> Word8
+    -- ^ Maximum number of inputs, estimated
+estimateMaxNumberOfInputsBase
+    EstimateMaxNumberOfInputsParams{..} (Quantity txSize) =
+    clamp $
+    max 0 (fromIntegral txSize - fixedSize)
+    `div`
+    (inputSize - fixedSize)
```

Comment:
Not sure about this computation here: this computes how many inputs can roughly fit in a given max size (minus the encoding overhead). Running this for the bridge using a max size of 8Kb, we find ~45. Though, if we take 45 inputs + witnesses, we are already above 8Kb and there's no room left for outputs :s !

Transactions are actually constructed from the outputs, which is "mostly" fixed number initially (we may have extra change created because of an overflow but that's an edge-case we can probably disregard here). 
In practice, this means that it's important to know the number of outputs too for making this computation (which is missing from the type-signature mentioned in the original ticket! My bad :s). 

For example, a transaction with 50 outputs (change excl.) will only leave room for ~23 inputs!

Reply:
OK, fixed.